WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverFragment.java

```diff
@@ -0,0 +1,242 @@
+                        mRecyclerView.post(new Runnable() {
+                            @Override
+                            public void run() {
+                                if (mRecyclerView.computeVerticalScrollRange() > mRecyclerView.getHeight()) {
+                                    mBottomButtonsShadow.setVisibility(View.VISIBLE);
+                                    mBottomButtonsContainer.setBackgroundResource(R.color.white);
+                                    mShareMediaBtn.setTextColor(getResources().getColor(R.color.blue_wordpress));
+                                    ViewUtils.setButtonBackgroundColor(getContext(), mShareMediaBtn,
+                                        R.style.WordPress_Button_Grey, R.attr.colorButtonNormal);
+                                } else {
+                                    mBottomButtonsShadow.setVisibility(View.GONE);
+                                    mBottomButtonsContainer.setBackground(null);
+                                    mShareMediaBtn.setTextColor(getResources().getColor(R.color.grey_dark));
+                                    ViewUtils.setButtonBackgroundColor(getContext(), mShareMediaBtn, R.style.WordPress_Button,
+                                        R.attr.colorButtonNormal);
+                                }
+                            }
+                        });
+                        mAdapter.findAndSelect(mLastUsedBlogLocalId);
+                    }
+                }
+            },
+            createHeaderHandler(),
+            null
+        );
+        mAdapter.setSingleItemSelectionEnabled(true);
+        return mAdapter;
+    }
+
+    private HeaderHandler createHeaderHandler() {
+        return new HeaderHandler() {
+            @Override
+            public ViewHolder onCreateViewHolder(LayoutInflater layoutInflater, ViewGroup parent,
+                boolean attachToRoot) {
+                return new LoginEpilogueFragment.HeaderViewHolder(layoutInflater.inflate(R.layout.login_epilogue_header, parent,
+                    false));
+            }
+
+            @Override
+            public void onBindViewHolder(ViewHolder holder, int numberOfSites) {
+                refreshAccountDetails((LoginEpilogueFragment.HeaderViewHolder) holder);
+            }
+        };
+    }
+
+    private void refreshAccountDetails(LoginEpilogueFragment.HeaderViewHolder holder) {
+        if (!isAdded()) {
+            return;
+        }
+        holder.update(getContext(), holder, mAfterLogin && mAccountStore.hasAccessToken(), mAccountStore.getAccount());
+        holder.mMySitesHeadingTextView.setText(R.string.share_intent_pick_site);
+        holder.mMySitesHeadingTextView.setVisibility(View.VISIBLE);
+    }
+
+    enum ShareAction {
+        SHARE_TO_POST(1, "new_post", EditPostActivity.class), SHARE_TO_MEDIA_LIBRARY(2, "media_library",
+            MediaBrowserActivity.class);
+
+        public final Class targetClass;
+        public final int id;
```

Comment:
I prefer using a custom field for enum identifier over ordinal/name because it's safe in all use-cases. 
* Ordinal is very tricky since the identifier can change when you swap two enum constants.
* Name is a bit tricky as well because it changes the id when you rename the enum constant.

On the other hand in this specific use-case, it doesn't really matter. It's just an internal representation and it's never stored into persistent storage or provided to another system. 

I'll use the name/valueof directly:).

Reply:
Indeed, when persistence is involved, using a custom `id` is the safe path to follow.

As you already noted, in this specific use case where any change to the order or the names doesn't practically matter, an `id` is redundant. Thanks for making the change!